thrust/system/cuda/detail/unique.h

```diff
@@ -794,6 +804,36 @@ unique(execution_policy<Derived> &policy,
   return cuda_cub::unique(policy, first, last, equal_to<input_type>());
 }
 
+
+template <typename BinaryPred>
+struct zip_adj_not_predicate {
+  template <typename TupleType>
+  bool __host__ __device__ operator()(TupleType&& tuple) {
+      return !binary_pred(thrust::get<0>(tuple), thrust::get<1>(tuple));
+  }
+  
+  BinaryPred binary_pred;
+};
+
+
+__thrust_exec_check_disable__
+template <class Derived,
+          class InputIt,
```

Comment:
Should be `ForwardIterator`, right?

Reply:
Right, that was copy-pasted from the definition above. Should I change it there as well?

Reply:
Yes, please do.

Reply:
done